discord/state.py

```diff
@@ -1681,31 +1681,50 @@ def parse_ready_supplemental(self, extra_data: gw.ReadySupplementalEvent) -> Non
     def parse_resumed(self, data: gw.ResumedEvent) -> None:
         self.dispatch('resumed')
 
-    def parse_passive_update_v1(self, data: gw.PassiveUpdateEvent) -> None:
-        # PASSIVE_UPDATE_V1 is sent for large guilds you are not subscribed to
-        # in order to keep their read and voice states up-to-date; it replaces CHANNEL_UNREADS_UPDATE
+    def parse_passive_update_v2(self, data: gw.PassiveUpdateEvent) -> None:
+        # PASSIVE_UPDATE_V2 is sent for large guilds you are not subscribed to
+        # in order to keep their members/read and voice states up-to-date; it replaces CHANNEL_UNREADS_UPDATE and PASSIVE_UPDATE_V1
         guild = self._get_guild(int(data['guild_id']))
         if not guild:
-            _log.debug('PASSIVE_UPDATE_V1 referencing an unknown guild ID: %s. Discarding.', data['guild_id'])
+            _log.debug('PASSIVE_UPDATE_V2 referencing an unknown guild ID: %s. Discarding.', data['guild_id'])
             return
 
-        for channel_data in data.get('channels', []):
+        for user_id in map(int, data.get('removed_voice_states', [])):
+            old = guild._voice_states.pop(user_id, None)
+            member = guild.get_member(user_id)
+            if member is None or old is None:
+                continue
+            new = copy.copy(old)
+            new.channel = None
+            self.dispatch('voice_state_update', member, old, new)
```

Comment:
```suggestion
        for user_id in map(int, data.get('removed_voice_states', [])):
            guild._voice_states.pop(user_id, None)
```
Events aren't dispatched for passive updates as they aren't real-time and these guilds aren't subscribed anyway